Update Build to use Qt 5.15.13 instead of Qt 5.15.12

notrege opened the pull request:
<!-- Lines like this one are comments and will not be shown in the final output. -->
<!-- Make sure that you have read the "Contributing" section of the README and also the notes in CodingStyle. -->
<!-- If you are a collaborator, please add labels and assign other collaborators for a review. -->

### Describe the pull request:
<!-- Replace [ ] with [x] to select options. -->
- [ ] Bug fix
- [ ] Functional change
- [ ] New feature
- [ ] Code cleanup
- [ X] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:
The current build using Qt 5.15.12 does not find some BLE devices when run on a Mac with Apple Silicon.   
An x86_64 MacOs binary built against Qt 5.15.13 will find and upload my G2 without any code changes.    I could not find a pre-built version of Qt 5.15.13 so I downloaded and built the source from https://download.qt.io/official_releases/qt/5.15/5.15.13.    After building on an Intel Mac and copying the entire app directory to an M2 Mac, I was able to connect and download 100 dives from a G2.

### Changes made:
No code or script changes - just a note in INSTALL about using Qt 5.15.13.
<!-- Ensure the test cases are updated if needed. -->

### Related issues:
<!-- Reference issues with #<issue-num>. -->
<!-- Write "Fixes #<issue-num" to notify Github that this PR fixes an issue. -->

### Additional information:
<!-- Include sample dive log or other relevant information to allow testing the change where feasible. -->

### Release note:
<!-- Describe if this change needs a release note present in CHANGELOG.md. -->
<!-- Also, please make sure to add the release note on top of the file CHANGELOG.md. -->

### Documentation change:
<!-- If this PR makes changes to user functionality, then the documentation has to be updated too. -->
<!-- Please, briefly outline here what has changed in terms of the user experience (UX). -->
<!-- If UX changes have been made, a maintainer should apply the 'needs-documentation-change' label. -->

### Mentions:
<!-- Mention users that you want to review your pull request with @<user-name>. Leave empty if not sure. -->

bstoeger commented on INSTALL:
```diff
@@ -320,7 +320,12 @@ brew install autoconf automake libtool pkg-config gettext
+preferred over Qt6.  
+
+If you plan to deploy your build to an Apple Silicon Mac, you may have better results
+results with Bluetooth connections if you install Qt5.15.13.  If Qt5.15.13 is not available
```
There is a duplicate 'results' here.

Otherwise, looks good to me.

mikeller commented:
Ha, this is a bit more 'barebones' than what I'd expected. 😝

In order to actually change anything I think we should explore if we can update the CICD Action that is building the MacOS image to use Qt 5.15.13.

From what I could find we are currently using a custom archive based on (presumably) Qt 5.15.2 to do this:

https://github.com/subsurface/subsurface/blob/3153a139b3c036a45839e8415541db71031160a8/.github/workflows/mac.yml#L39-L45

Not sure if the contents of the archive are the equivalent of what is downloadable from Qt, but probably worth a try to replace them and see what happens?

notrege commented:
Sorry - all new territory for me.  As usual, thanks for help.

Who owns this:
https://f002.backblazeb2.com/file/Subsurface-Travis/ssrf-Qt5.15.2.tar.xz?

If we can put 5.15.13 there and change
GITHUB_WORKSPACE}/Qt5.15.2/5.15.2/clang_64 it should work.

I can provide a 5.15.13 tarball if that helps, but I don't have a spot to
host it.

thanks,
Jim

On Sun, May 5, 2024, 09:48 Michael Keller ***@***.***> wrote:

> Ha, this is a bit more 'barebones' than what I'd expected. 😝
>
> In order to actually change anything I think we should explore if we can
> update the CICD Action that is building the MacOS image to use Qt 5.15.13.
>
> From what I could find we are currently using a custom archive based on
> (presumably) Qt 5.15.2 to do this:
>
>
> https://github.com/subsurface/subsurface/blob/3153a139b3c036a45839e8415541db71031160a8/.github/workflows/mac.yml#L39-L45
>
> Not sure if the contents of the archive are the equivalent of what is
> downloadable from Qt, but probably worth a try to replace them and see what
> happens?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/subsurface/subsurface/pull/4177#issuecomment-2094690576>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHV6KUQHOX3QWJQ5PBFHUF3ZAXW4ZAVCNFSM6AAAAABHHNQTZKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDAOJUGY4TANJXGY>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

notrege commented:
I have not found anywhere other than brew to download a pre-built version of Qt.    There are both online and offline installers for Qt, but I haven’t found either for 5.15.13.     

I downloaded from https://download.qt.io/archive/qt/5.15/5.15.13/single/qt-everywhere-opensource-src-5.15.13.tar.xz and built it.    Theoretically we could make that part of the build, but that would take a long, long time to build.    

The only options I can see are tarring up the output from either a brew installation or a build from source.    Again, this is all new to me so if someone else has a better idea, please don’t hold back.     I seem to remember a comment someone (Dirk?) made about problems using the brew installation - I think it was something about packaging the framework, but I don’t remember the details.   

Additionally, the Macs I have available are an Intel MacBook Pro running Catalina and a M2 Macbook running Sonoma.    The Catalina Macbook can’t be upgraded and I can’t brew install Qt beyond 5.15.7 so the only option I can offer would be to tar up the output from my build.   Am happy to provide that knowing that it’s a small change to rollback mac.yml if it goes south.


> On May 5, 2024, at 10:24, Michael Keller ***@***.***> wrote:
> 
> 
> Who owns this: f002.backblazeb2.com/file/Subsurface-Travis/ssrf-Qt5.15.2.tar.xz <https://f002.backblazeb2.com/file/Subsurface-Travis/ssrf-Qt5.15.2.tar.xz>?
> 
> I think Backblaze is a file hosting service akin to Dropbox - so the answer is probably 'whoever put it there'.
> 
> If we can put 5.15.13 there and change GITHUB_WORKSPACE}/Qt5.15.2/5.15.2/clang_64 it should work. I can provide a 5.15.13 tarball if that helps, but I don't have a spot to host it.
> 
> I was wondering if it will be possible to just pull the Qt 5.15.13 package from downloads.qt.io, instead of building our own tarball.
> If we have to build our own I can probably find a spot to host it.
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/subsurface/subsurface/pull/4177#issuecomment-2094702956>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AHV6KUX3ZFEKYWTBHMAVHNDZAX3DTAVCNFSM6AAAAABHHNQTZKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDAOJUG4YDEOJVGY>.
> You are receiving this because you authored the thread.
>

dirkhh commented:
That backblaze bucket is of course mine. 
Reason for having it there is that others charge you for egress and with all the build we are doing that was asking up quickly to some real money. 
I'm happy to put other binaries there

mikeller commented:
> The only options I can see are tarring up the output from either a brew installation or a build from source. Again, this is all new to me so if someone else has a better idea, please don’t hold back. I seem to remember a comment someone (Dirk?) made about problems using the brew installation - I think it was something about packaging the framework, but I don’t remember the details.

Ah I get you - our own custom tarball are prebuilt binaries, whereas the download from Qt are the sources...

I wish we were building this on docker, or there was a mechanism like docker on MacOS that allowed you to run stuff in pre-built images. We are working around similar issues for the Windows and Android builds by running long-running, but rarely chanbing preparation tasks as part of building a custom docker image, and then simply run our actual builds inside a docker container running this image, so that everything is 'just there'.

dirkhh commented:
Creating the Mac download is an incredibly frustrating task. You first have the painful and slow build, and the the tedious task of pruning as much as possible from the binaries so that things still build, but that the binaries aren't ginormous.
Jim sent me a link to his attempt to get this to work and I put this in the Backblaze bucket here:

https://f002.backblazeb2.com/file/Subsurface-Travis/ssrf-Qt5.15.13.tar.xz

notrege commented:
That’s pretty much how it works.    You can use Qt online or offline
installers to install a recent version of Qt and even some of the older
versions, but when they decide to archive a version it disappears from the
online installer.   If we could install with the online or offline
installer we would still either have to run an install from
workflows/mac.yml or install and tar up the output, but  for some odd
reason, the last offline installer at https://www.qt.io/offline-installers
is for Qt 5.12 and if you follow the “Qt 5.15.x Source Packages” link it
defaults to Qt 5.15.2, but there are other versions at
https://download.qt.io/official_releases/qt/5.15/.

I’m not thrilled about building it and using that going forward, but I
don’t see an alternative.  If something goes wrong, the only thing to
revert will be the yml file so it's pretty low risk.  I'll create a pull
request for that and cross my fingers.

On May 5, 2024, at 20:27, Michael Keller ***@***.***> wrote:


The only options I can see are tarring up the output from either a brew
installation or a build from source. Again, this is all new to me so if
someone else has a better idea, please don’t hold back. I seem to remember
a comment someone (Dirk?) made about problems using the brew installation -
I think it was something about packaging the framework, but I don’t
remember the details.

Ah I get you - our own custom tarball are prebuilt binaries, whereas the
download from Qt are the sources...

I wish we were building this on docker, or there was a mechanism like
docker on MacOS that allowed you to run stuff in pre-built images. We are
working around similar issues for the Windows and Android builds by running
long-running, but rarely chanbing preparation tasks as part of building a
custom docker image, and then simply run our actual builds inside a docker
container running this image, so that everything is 'just there'.

—
Reply to this email directly, view it on GitHub
<https://github.com/subsurface/subsurface/pull/4177#issuecomment-2094920911>,
or unsubscribe
<https://github.com/notifications/unsubscribe-auth/AHV6KUWBPBCCGNU4FX2RVSLZA2BY3AVCNFSM6AAAAABHHNQTZKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDAOJUHEZDAOJRGE>
.
You are receiving this because you modified the open/close state.Message
ID: ***@***.***>

notrege commented:
Looks like you beat me to the change in mac.yml with https://github.com/subsurface/subsurface/pull/4180  - thanks Michael. 

> On May 6, 2024, at 06:49, Jim E ***@***.***> wrote:
> 
> That’s pretty much how it works.    You can use Qt online or offline installers to install a recent version of Qt and even some of the older versions, but when they decide to archive a version it disappears from the online installer.   If we could install with the online or offline installer we would still either have to run an install from workflows/mac.yml or install and tar up the output, but  for some odd reason, the last offline installer at https://www.qt.io/offline-installers  is for Qt 5.12 and if you follow the “Qt 5.15.x Source Packages” link it defaults to Qt 5.15.2, but there are other versions at https://download.qt.io/official_releases/qt/5.15/.
> 
> I’m not thrilled about building it and using that going forward, but I don’t see an alternative.  If something goes wrong, the only thing to revert will be the yml file so it's pretty low risk.  I'll create a pull request for that and cross my fingers.
> 
> 
>> On May 5, 2024, at 20:27, Michael Keller ***@***.*** ***@***.***>> wrote:
>> 
>> 
>> The only options I can see are tarring up the output from either a brew installation or a build from source. Again, this is all new to me so if someone else has a better idea, please don’t hold back. I seem to remember a comment someone (Dirk?) made about problems using the brew installation - I think it was something about packaging the framework, but I don’t remember the details.
>> 
>> Ah I get you - our own custom tarball are prebuilt binaries, whereas the download from Qt are the sources...
>> 
>> I wish we were building this on docker, or there was a mechanism like docker on MacOS that allowed you to run stuff in pre-built images. We are working around similar issues for the Windows and Android builds by running long-running, but rarely chanbing preparation tasks as part of building a custom docker image, and then simply run our actual builds inside a docker container running this image, so that everything is 'just there'.
>> 
>> —
>> Reply to this email directly, view it on GitHub <https://github.com/subsurface/subsurface/pull/4177#issuecomment-2094920911>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AHV6KUWBPBCCGNU4FX2RVSLZA2BY3AVCNFSM6AAAAABHHNQTZKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDAOJUHEZDAOJRGE>.
>> You are receiving this because you modified the open/close state.
>> 
>

mikeller commented:
> Looks like you beat me to the change in mac.yml with #4180 - thanks Michael.

😆

The good news is that it seems to be working! You should be able to download https://github.com/subsurface/subsurface/actions/runs/8965203901/artifacts/1475593686
now and check if this is working on Apple silicon.

@dirkhh: For some reason ssrf-Qt5.15.13.tar.xz is in fact gzipped - which is why the `curl` in the GitHub Action looks so confused and wrong.
It's probably easiest if you decompress, recompress and then re-upload this, and then ping me when it has been changed so that I can update the action.

notrege commented:
Sorry about that - tar czf is a bit of a force of habit.    Unzipped MacOS-artifacts.zip, installed on my M2 and it seems to work fine.   Found my G2, connected, and correctly reported no new dives - happy days.

￼
￼

thanks for all the help with this,
jim

> On May 6, 2024, at 10:43, Michael Keller ***@***.***> wrote:
> 
> 
> Looks like you beat me to the change in mac.yml with #4180 <https://github.com/subsurface/subsurface/pull/4180> - thanks Michael.
> 
> 😆
> 
> The good news is that it seems to be working! You should be able to download https://github.com/subsurface/subsurface/actions/runs/8965203901/artifacts/1475593686
> now and check if this is working on Apple silicon.
> 
> @dirkhh <https://github.com/dirkhh>: For some reason ssrf-Qt5.15.13.tar.xz is in fact gzipped - which is why the curl in the GitHub Action looks so confused and wrong.
> It's probably easiest if you decompress, recompress and then re-upload this, and then ping me when it has been changed so that I can update the action.
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/subsurface/subsurface/pull/4177#issuecomment-2095577625>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AHV6KUSQAGLFBUBYOYYWXM3ZA5GCVAVCNFSM6AAAAABHHNQTZKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDAOJVGU3TONRSGU>.
> You are receiving this because you modified the open/close state.
>

dirkhh commented:
I recompressed the file and changed it to have a name consistent with what we had before:

https://f002.backblazeb2.com/file/Subsurface-Travis/Qt5.15.13-mac.tar.xz

notrege commented:
Thanks Dirk - sorry for making extra work.

I think the -mac was added to the name with curl, but it doesn't matter.

It might make sense to drop the version number with curl so the tar line
doesn't have to be updated, but the version number still needs to be
changed for the cmake root unless, of course, we drop the prefix when
creating the tarball.

I'm just happy that it works now.

On Tue, May 7, 2024, 00:49 Dirk Hohndel ***@***.***> wrote:

> I recompressed the file and changed it to have a name consistent with what
> we had before:
>
> https://f002.backblazeb2.com/file/Subsurface-Travis/Qt5.15.13-mac.tar.xz
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/subsurface/subsurface/pull/4177#issuecomment-2097109635>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHV6KUTKVH7SRCQWUGFJ3RTZBAJKPAVCNFSM6AAAAABHHNQTZKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDAOJXGEYDSNRTGU>
> .
> You are receiving this because you modified the open/close state.Message
> ID: ***@***.***>
>

mikeller commented:
> It might make sense to drop the version number with curl so the tar line doesn't have to be updated, but the version number still needs to be changed for the cmake root unless, of course, we drop the prefix when creating the tarball.

I would actually quite like to keep the version number - to me having a tiny bit of update work whenever we decide we want to switch a library version, and getting a predictable and repeatable build, is preferrable to using 'mystery box' componentry in return for a bit less work.

notrege commented:
What I was suggesting is keeping the version number in the URL, especially
since we may need to revert, but dropping it everywhere else

On Tue, May 7, 2024, 07:41 Michael Keller ***@***.***> wrote:

> It might make sense to drop the version number with curl so the tar line
> doesn't have to be updated, but the version number still needs to be
> changed for the cmake root unless, of course, we drop the prefix when
> creating the tarball.
>
> I would actually quite like to keep the version number - to me having a
> tiny bit of update work whenever we decide we want to switch a library
> version, and getting a predictable and repeatable build, is preferrable to
> using 'mystery box' componentry in return for a bit less work.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/subsurface/subsurface/pull/4177#issuecomment-2097559241>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHV6KUXNDNCC26XPOSFJ663ZBBZQPAVCNFSM6AAAAABHHNQTZKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDAOJXGU2TSMRUGE>
> .
> You are receiving this because you modified the open/close state.Message
> ID: ***@***.***>
>

mikeller commented:
> What I was suggesting is keeping the version number in the URL, especially since we may need to revert, but dropping it everywhere else

It's mostly gone, except for the URL, and the path. So we can probably strip out the path the next time we touch this.
